var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -184,6 +194,7 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     patch('xlf_fix-dup-petscfecreate.patch', when='@3.11.0')
     patch('disable-DEPRECATED_ENUM.diff', when='@3.14.1 +cuda')
 
+    depends_on('gmake', type='build', when='+suite-sparse')
```

Comment:
PETSc build requires gnumake - so `when='+suite-sparse'` should be removed.

Reply:
Ok I'll leave it like that.
It was a guess from:
`config/BuildSystem/config/packages/SuiteSparse.py:46:      raise RuntimeError('SuiteSparse buildtools require GNUMake. Use --with-make=gmake or --download-make')`
But probably other packages needs it.

Reply:
Ah - ok. This check might have been added to petsc configure before petsc switched to requiring gnumake [so we could remove this check for suite-sparse from petsc..]

And that check is for `--download-suitesparse` - which should never get used when building via spack..

Also we haven't seen a failure in petsc build so far without this dependency - so  perhaps spack already defaults to using gnumake?

Reply:
> Also we haven't seen a failure in petsc build so far without this dependency - so perhaps spack already defaults to using gnumake?

Actually, I added the dependency based on a failure. Though I now wonder if it was a problem with `gmake` or something else since `gmake` is checked system wide and CentOS `make` comes with `gmake`... Anyway, I am currently rebuilding `PETSc` with a bare CentOS linux without this dependency, I'll tell you the result as soon as I get it.

Reply:
It works without `gmake` (branch `develop`), so I removed it.